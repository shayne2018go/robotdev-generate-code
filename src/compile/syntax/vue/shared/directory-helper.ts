/**
 * 根据目录获取路径
 * @param directories 目录
 * @param id id
 * @returns
 */
export function getPathByDirectories(directories: CodeSchema.Directory[], id: string): CodeSchema.Directory[] {
  const dirPath: CodeSchema.Directory[] = [];

  let curId: string | null = id;

  while (curId) {
    const current = directories.find((d) => d.id === curId);
    if (current) {
      dirPath.unshift(current);
      curId = current.parentId;
    } else {
      break;
    }
  }

  return dirPath;
}
