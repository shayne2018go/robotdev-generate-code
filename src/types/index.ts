export * from './ICodeSchema';

export * from './DBSchema';

export * from './DBWSchema';
