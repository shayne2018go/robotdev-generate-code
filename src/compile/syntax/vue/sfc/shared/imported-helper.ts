import { relative } from '@/utils/node';
import * as t from '@babel/types';

export interface ImportedParsing {
  importPath: string;
  specifiers: (t.ImportSpecifier | t.ImportDefaultSpecifier | t.ImportNamespaceSpecifier)[];
}

export function parsingImportDeclaration(
  source: GlobalContext.Source,
  currentDir: string
): ImportedParsing | undefined {
  const importPath = source.packageName
    ? source.packageName
    : source.filePath
    ? relative(currentDir, source.filePath)
    : '';

  if (!importPath) {
    return;
  }

  const specifiers = [] as ImportedParsing['specifiers'];
  if (source.exportNamespace) {
    specifiers.push(t.importNamespaceSpecifier(t.identifier(source.exportNamespace)));
  } else if (source.exportName) {
    const alias = source.alias || source.exportName;
    specifiers.push(t.importSpecifier(t.identifier(alias!), t.identifier(source.exportName)));
  }

  return {
    importPath,
    specifiers,
  };
}

export function parsingImportedParsingArray(
  data: (GlobalContext.Function | GlobalContext.Component | GlobalContext.Api)[],
  currentDir: string
) {
  const importedParsingArray = data.reduce((pre, cur) => {
    if (!cur.source) {
      return pre;
    }
    const importedParsing = parsingImportDeclaration(cur.source, currentDir);

    if (!importedParsing) {
      return pre;
    }

    const isExistImportPath = pre.find((p) => p.importPath === importedParsing?.importPath);

    if (isExistImportPath) {
      const noExistSpecifiers = importedParsing.specifiers.filter(
        (spec) => !isExistImportPath.specifiers.find((espec) => isEqualSpecifier(espec, spec))
      );
      isExistImportPath.specifiers = isExistImportPath.specifiers.concat(noExistSpecifiers);
    } else {
      pre.push(importedParsing);
    }

    return pre;
  }, [] as ImportedParsing[]);
  return importedParsingArray;
}

export function isEqualSpecifier(
  left: ImportedParsing['specifiers'][number],
  right: ImportedParsing['specifiers'][number]
) {
  if (left.type !== right.type || left.local.name !== right.local.name) {
    return false;
  }
  if (t.isImportSpecifier(left) && t.isImportSpecifier(right)) {
    if (t.isIdentifier(left.imported) && t.isIdentifier(right.imported)) {
      return left.imported.name === right.imported.name;
    } else if (t.isStringLiteral(left.imported) && t.isStringLiteral(right.imported)) {
      return left.imported.value === right.imported.value;
    }
  } else if (t.isImportNamespaceSpecifier(left) && t.isImportNamespaceSpecifier(right)) {
    return left.local.name === right.local.name;
  }

  return false;
}

export function getCurrentDirByPath(path: string): string {
  return path.split('/').slice(0, -1).join('/');
}
