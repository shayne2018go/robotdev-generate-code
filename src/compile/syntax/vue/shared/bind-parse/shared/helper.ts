import { DBWSchema } from '@/types/DBWSchema';
import { tools } from '@/utils/tools';

export const isRdData = (data: any): data is DBWSchema.RdData => {
  if (tools.dataType.isObject(data) && data.type === 'data' && data.mode) {
    return true;
  }
  return false;
};
export const rdDataisCustom = (data: DBWSchema.RdData): data is DBWSchema.RdData_Custom => {
  return data.mode === 'custom';
};
export const rdDataIsBind = (
  data: DBWSchema.RdData
): data is
  | DBWSchema.RdData_GetVar
  | DBWSchema.RdData_GetApiData
  | DBWSchema.RdData_GetParam
  | DBWSchema.RdData_GetEventData
  | DBWSchema.RdData_GetSlotData
  | DBWSchema.RdData_GetEachData
  | DBWSchema.RdData_GetModelData
  | DBWSchema.RdData_GetCmptPropData
  | DBWSchema.RdData_TableData => {
  return [
    'getVar',
    'getApiData',
    'getParam',
    'getEventData',
    'getSlotData',
    'getEachData',
    'getModelData',
    'getCmptPropData',
    'tableData',
  ].includes(data.mode);
};

export const genRdData = (
  type: DBWSchema.RdData_Custom['args']['type'],
  value: any,
  multiple?: boolean
): DBWSchema.RdData_Custom => {
  return {
    id: '',
    modeId: '',
    type: 'data',
    mode: 'custom',
    args: {
      type: type,
      multiple: multiple === true,
      value: value,
    },
  };
};

export const literalToRdData_Custom = (data: any): DBWSchema.RdData_Custom => {
  const type = tools.dataType.getType(data);
  switch (type) {
    case 'string': {
      return genRdData('text', data);
    }
    case 'number': {
      return genRdData('number', data);
    }
    case 'whether': {
      return genRdData('whether', data);
    }
    case 'object': {
      const objList: { propId: string; value: any }[] = [];
      Object.entries(data).forEach((item) => {
        objList.push({
          propId: item[0],
          value: item[1],
        });
      });
      return genRdData('module', objList);
    }
    case 'array': {
      let type: DBWSchema.RdData_Custom['args']['type'] = '';
      const list: any[] = [];
      data.forEach((item: any, index: number) => {
        const itemRdData = isRdData(item) ? item : literalToRdData_Custom(item);
        if (!type && rdDataisCustom(itemRdData)) {
          type = itemRdData.args.type;
        }
        list.push(itemRdData);
      });
      return genRdData(type, list, true);
    }
  }
  throw new Error('数据类型有误');
};
