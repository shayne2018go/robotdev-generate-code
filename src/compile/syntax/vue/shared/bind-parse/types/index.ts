import {
  ArrayExpression,
  AssignmentExpression,
  BinaryExpression,
  BooleanLiteral,
  CallExpression,
  DecimalLiteral,
  Identifier,
  IfStatement,
  MemberExpression,
  NullLiteral,
  NumericLiteral,
  ObjectExpression,
  OptionalMemberExpression,
  StringLiteral,
} from '@babel/types';
import { CompileCurrentCtx } from '../../../compilePages';
import { genVarName } from '../../helper';

export type ActionAst = CallExpression | AssignmentExpression;
export type ActionsAst = (CallExpression | AssignmentExpression)[];

export type BindAst =
  | OptionalMemberExpression
  | MemberExpression
  | ArrayExpression
  | StringLiteral
  | NumericLiteral
  | BooleanLiteral
  | ObjectExpression
  | DecimalLiteral
  | CallExpression
  | NullLiteral
  | Identifier
  | BinaryExpression;

export type BindRdData =
  | CodeSchema.DataValue_GetVar
  | CodeSchema.DataValue_GetApiData
  | CodeSchema.DataValue_GetParam
  | CodeSchema.DataValue_GetEventData
  | CodeSchema.DataValue_GetSlotData
  | CodeSchema.DataValue_GetEachData
  | CodeSchema.DataValue_GetArguments
  // | CodeSchema.DataValue_GetModelData
  | CodeSchema.DataValue_GetCmptPropData
  | CodeSchema.DataValue_TableData;

export type LiteralAst =
  | ArrayExpression
  | StringLiteral
  | NumericLiteral
  | BooleanLiteral
  | ObjectExpression
  | DecimalLiteral
  | CallExpression
  | Identifier;

export type BindParseCtx<T extends CodeSchema.Page | CodeSchema.Component = CodeSchema.Page> = {
  global: CompileCurrentCtx['global'];
  scope: CompileCurrentCtx['scope'] & {
    node?: CodeSchema.ComponentNode;
    prop?: CodeSchema.Property;
    event?: CodeSchema.Event;
    actions?: {
      genVarName: ReturnType<typeof genVarName>;
      map: {
        [id: CodeSchema.Action['id']]: {
          data: CodeSchema.Action;
          protocol: CodeSchema.Action_Protocol;
          varName: string;
          parametersVarNames: {
            [id: CodeSchema.Action_Protocol['parameters'][number]['id']]: {
              protocol: CodeSchema.Action_Protocol['parameters'][number];
              varName: string;
              memberVarNames: {
                [id: CodeSchema.Property_Protocol['id']]: {
                  protocol: CodeSchema.Property_Protocol;
                  varName: string;
                };
              };
            };
          };
        };
      };
    };
  };
};

export type AccessPath = {
  type: 'object' | 'array';
  key: string | number;
}[];

export type SplitProps = {
  key: string;
  value: BindAst | undefined;
}[];

export type ReturnRef = {
  type: 'split' | 'ast' | 'error';
  value?: ActionAst | ActionsAst | IfStatement | BindAst | SplitProps | undefined;
};
