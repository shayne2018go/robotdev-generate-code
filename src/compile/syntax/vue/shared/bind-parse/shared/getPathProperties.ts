import { searchModulePath } from '../../searchPath';
import { BindParseCtx } from '../types';
import { getEachIndexVarName, getEachItemVarName, isRdData, rdDataIsBind } from './helper';

const genApiResProp = (api: CodeSchema.Api_Protocol) => {
  return {
    id: api.id,
    key: api.key,
    types: [
      {
        type: 'module',
        rules: {
          properties: [
            {
              id: 'status',
              key: 'status',
              // name: '请求是否成功',
              types: [{ type: 'whether' }],
            },
            {
              id: 'httpcode',
              key: 'httpcode',
              // name: '请求状态码（400为未登录）',
              types: [{ type: 'number' }],
            },
            {
              id: 'data',
              key: 'data',
              // name: '请求成功后数据',
              types: [
                {
                  type: 'module',
                  multiple: true,
                  rules: {
                    properties: [
                      {
                        id: 'code',
                        key: 'code',
                        types: [{ type: 'text' }],
                        // name: '业务状态码',
                      },
                      {
                        id: 'message',
                        key: 'message',
                        types: [{ type: 'text' }],
                        // name: '业务提示语',
                      },
                      {
                        id: 'data',
                        key: 'data',
                        types: [{ type: 'module', rules: { properties: api.response.body } }],
                        // name: '业务数据',
                      },
                    ],
                  },
                },
              ],
            },
            {
              id: 'message',
              key: 'message',
              // name: '提示语',
              types: [{ type: 'text' }],
            },
          ],
        },
      },
    ],
    // name: '请求是否成功',
  } as CodeSchema.Property_Protocol;
};

const genEachModuleType = (varName: string, types: CodeSchema.PropertyType_Protocol[]) => {
  const itemTypes = types.map((type) => {
    return { ...type, multiple: false };
  });
  return {
    type: 'module',
    rules: {
      properties: [
        {
          id: 'item',
          key: getEachItemVarName(varName),
          types: itemTypes,
          // name: '每次循环得到的数据',
        },
        {
          id: 'index',
          key: getEachIndexVarName(varName),
          types: [{ type: 'index' }],
          // name: '索引',
        },
      ],
    },
  } as CodeSchema.PropertyType_Protocol;
};

const getBindPathProperties = {
  getVar(
    ctx: BindParseCtx,
    data: CodeSchema.DataValue_GetVar['args'],
    isRoot = false,
    pathLastIsModuleMultiple?: boolean
  ) {
    if (!data?.id) {
      return;
    }
    const define = ctx.scope.current.variablesStore.findId(data.id);
    if (!define) {
      throw new Error('getVar方法的定义获取失败');
    }
    const accessPath = searchModulePath(define.data.types, data.path || [], pathLastIsModuleMultiple); // 路径中的每个属性名
    if (isRoot) {
      return [define.data, ...accessPath];
    }
    return accessPath;
  },
  getApiData(
    ctx: BindParseCtx,
    data: CodeSchema.DataValue_GetApiData['args'],
    isRoot = false,
    pathLastIsModuleMultiple?: boolean
  ) {
    if (!data?.id) {
      return;
    }
    const define = ctx.global.apisStore.getApi(data.id);
    if (!define) {
      throw new Error('getApiData方法的定义获取失败');
    }
    const keyId = data.path?.[1];
    if (!keyId) {
      return;
    }
    const apiResProp = genApiResProp(define?.data.protocol || []);
    const accessPath = searchModulePath(apiResProp.types, data.path || [], pathLastIsModuleMultiple); // 路径中的每个属性名
    if (isRoot) {
      return [apiResProp, ...accessPath];
    }
    return accessPath;
  },
  getParam(
    ctx: BindParseCtx,
    data: CodeSchema.DataValue_GetParam['args'],
    isRoot = false,
    pathLastIsModuleMultiple?: boolean
  ) {
    if (!data?.id) {
      return;
    }
    const define = ctx.global.pagesStore.getQuery(ctx.scope.current.data.id, data.id);
    if (!define) {
      throw new Error('getParam方法的定义获取失败');
    }
    const accessPath = searchModulePath(define.data.types || [], data.path || [], pathLastIsModuleMultiple); // 路径中的每个属性名
    if (isRoot) {
      return [define.data, ...accessPath];
    }
    return accessPath;
  },
  getCmptPropData(
    ctx: BindParseCtx,
    data: CodeSchema.DataValue_GetCmptPropData['args'],
    isRoot = false,
    pathLastIsModuleMultiple?: boolean
  ) {
    if (!data?.id) {
      return;
    }

    const define = ctx.scope.current.propsStore.findId(data.id);
    if (!define) {
      throw new Error('getCmptPropData方法的定义获取失败');
    }
    const accessPath = searchModulePath(define?.data.types || [], data.path || [], pathLastIsModuleMultiple); // 路径中的每个属性名
    if (isRoot) {
      return [define.data, ...accessPath];
    }
    return accessPath;
  },
  getSlotData(
    ctx: BindParseCtx,
    data: CodeSchema.DataValue_GetSlotData['args'],
    isRoot = false,
    pathLastIsModuleMultiple?: boolean
  ) {
    if (!data?.id) {
      return;
    }
    const node = ctx.scope.current.nodesStore.find(data.id);
    if (!node || !ctx.scope.node?.id) {
      return;
    }
    let slotNode: CodeSchema.ComponentNode | undefined;
    const slotAffiliationCmpt = ctx.scope.current.nodesStore.parentOne(ctx.scope.node?.id, (item) => {
      if (item.store?.component?.data.protocol.key === 'tpl') {
        if (item.data.parentId) {
          if (item.data.parentId === data.id) {
            slotNode = item.data;
          }
        } else {
          throw new Error('tpl节点没有parentId');
        }
      }
      return item.data.id === data.id;
    })?.component;
    if (!slotNode?.slotId) {
      throw new Error('slotNode未取到');
    }
    const propId = data.path?.[0];
    if (!propId) {
      return;
    }

    const slot = slotAffiliationCmpt?.members.slotsStore.findId(slotNode.slotId);
    if (!slot) {
      throw new Error('数据异常，插槽节点的属性的id没找到！');
    }
    const slotProp = slot.members.properties?.find(propId);
    if (!slotProp) {
      throw new Error('数据异常，插槽节点的属性的下的参数没有没找到相应属性！');
    }

    const accessPath = searchModulePath(slotProp.data.types || [], data.path?.slice(1) || [], pathLastIsModuleMultiple); // 路径中的每个属性名
    if (isRoot) {
      return [slotProp.data, ...accessPath];
    }
    return accessPath;
  },
  getEachData(ctx: BindParseCtx, data: CodeSchema.DataValue_GetEachData['args'], isRoot = false) {
    if (!data?.id) {
      return;
    }
    const node = ctx.scope.current.nodesStore.find(data.id);
    if (!node) {
      return;
    }
    const eachPropDataId = node.component?.members.propsStore.findKey('data')?.data.id;
    if (!eachPropDataId) {
      throw new Error('数据异常，循环节点的data属性的id没找到！');
    }
    const bindData = node.propsStore.query(eachPropDataId).value;
    if (!bindData) {
      return;
    }

    if (isRdData(bindData) && rdDataIsBind(bindData)) {
      const bindPathProperties = getBindPathProperties.getAnyData(ctx, bindData, true);
      if (!bindPathProperties?.length) {
        return;
      }
      if (!isRoot) {
        return bindPathProperties;
      }
      const last = bindPathProperties[bindPathProperties?.length - 1];

      const types = [genEachModuleType(node.varName, last.types)];
      const accessPath = searchModulePath(types || [], data?.path || []); // 路径中的每个属性名
      return accessPath;
    }
    return;
  },
  getAnyData(
    ctx: BindParseCtx,
    data: CodeSchema.DataValue | CodeSchema.Action,
    isRoot = false,
    pathLastIsModuleMultiple?: boolean
  ): CodeSchema.Property_Protocol[] | undefined {
    if (!isRdData(data)) {
      return [];
    }
    switch (data.mode) {
      case 'getEachData': {
        return getBindPathProperties.getEachData(ctx, (data as CodeSchema.DataValue_GetEachData).args, isRoot);
      }
      case 'getApiData': {
        return getBindPathProperties.getApiData(
          ctx,
          (data as CodeSchema.DataValue_GetApiData).args,
          isRoot,
          pathLastIsModuleMultiple
        );
      }
      case 'getVar': {
        return getBindPathProperties.getVar(
          ctx,
          (data as CodeSchema.DataValue_GetVar).args,
          isRoot,
          pathLastIsModuleMultiple
        );
      }
      case 'getParam': {
        return getBindPathProperties.getParam(
          ctx,
          (data as CodeSchema.DataValue_GetParam).args,
          isRoot,
          pathLastIsModuleMultiple
        );
      }
      case 'getCmptPropData': {
        return getBindPathProperties.getCmptPropData(
          ctx,
          (data as CodeSchema.DataValue_GetCmptPropData).args,
          isRoot,
          pathLastIsModuleMultiple
        );
      }
      case 'getSlotData': {
        return getBindPathProperties.getSlotData(
          ctx,
          (data as CodeSchema.DataValue_GetSlotData).args,
          isRoot,
          pathLastIsModuleMultiple
        );
      }
    }
    return [];
  },
};

export const getPathProperties = (
  ctx: BindParseCtx,
  data: CodeSchema.DataValue | CodeSchema.Action,
  pathLastIsModuleMultiple?: boolean
) => {
  return getBindPathProperties.getAnyData(ctx, data, true, pathLastIsModuleMultiple);
};

export const getPathPropertieKeys = (
  ctx: BindParseCtx,
  data: CodeSchema.DataValue | CodeSchema.Action,
  pathLastIsModuleMultiple?: boolean
) => {
  return getBindPathProperties.getAnyData(ctx, data, true, pathLastIsModuleMultiple)?.map((item) => item.key);
};
