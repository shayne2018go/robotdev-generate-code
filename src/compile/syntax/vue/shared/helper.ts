import { INode } from '@/types/view';
import { DBSchema } from '@/types';
import { VueTypes } from '../types/vue';
import { BUILT_IN_PACKAGES } from '../const/config';

// 出现次数统计工具
export const occurrenceNumber = () => {
  const cache: { [key: string]: number } = {};

  return (varName: string) => {
    if (typeof cache[varName] !== 'number') {
      cache[varName] = 0;
    } else {
      cache[varName]++;
    }
    return cache[varName];
  };
};

export const genVarName = () => {
  const occurrenceNumberHandler = occurrenceNumber();
  return (varName: string) => {
    const number = occurrenceNumberHandler(varName);
    return varName + (number && number > 0 ? `_${number + 1}` : '');
  };
};

export function outerNode(node: INode) {
  return node.packageId && !BUILT_IN_PACKAGES.includes(node.packageId);
}

/**
 * 获取节点中的组件依赖
 * @param nodes
 * @param componentsDependencies
 * @returns
 */
function getNodesComponentDependencies(
  nodes: INode[],
  componentsDependencies: VueTypes.Component[]
): Map<string, VueTypes.Component> {
  const outerNodes = getOuterNodes(nodes);
  const currentPageComponentDependencies = componentsDependencies.filter((cd) =>
    outerNodes.map((on) => on.tagId).includes(cd.id)
  );
  const map = new Map<string, VueTypes.Component>();
  currentPageComponentDependencies.forEach((cpcd) => {
    map.set(cpcd.id, cpcd);
  });
  return map;
}

function getOuterNodes(nodes: Array<INode>) {
  const outerNodes = nodes.filter((node) => {
    return node.packageId && !BUILT_IN_PACKAGES.includes(node.packageId);
  });
  return outerNodes;
}

/**
 * 获取行为中的行为依赖
 * @param actions
 */
function getActionDependencies(actions: DBSchema.Action[]) {}

export { getNodesComponentDependencies, getActionDependencies };
