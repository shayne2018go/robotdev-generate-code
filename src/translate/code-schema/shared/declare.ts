import { DBWSchema } from '@/types';
import { Identifier } from '@/types/code-schema/Identifier';

function preprocessDeclare(rdProp: DBWSchema.RdDecl): Identifier;
function preprocessDeclare(rdProp: DBWSchema.RdDefineProp): Identifier;
function preprocessDeclare(rdProp: DBWSchema.RdComponentDefineProp): Identifier;
function preprocessDeclare(
  rdProp: DBWSchema.RdDecl | DBWSchema.RdDefineProp | DBWSchema.RdComponentDefineProp
): Identifier {
  return {
    id: rdProp.id,
    name: rdProp.name!,
    key: rdProp.key!,
    desc: rdProp.desc,
    types: rdProp.types,
    extendPlatform: rdProp.extendPlatform,
  };
}

export { preprocessDeclare };
