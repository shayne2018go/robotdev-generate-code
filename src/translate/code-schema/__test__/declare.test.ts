import { DBWSchema } from '@/types';
import { describe, expect, it } from 'vitest';
import { preprocessDeclare } from '../shared/declare';

describe('declare', () => {
  it('preprocessDeclare text', () => {
    const rdProp = {
      id: '641a7052cf44f6204642b6bf',
      key: 'class',
      types: [
        {
          type: 'text',
        },
      ],
      name: '布局样式',
      desc: '容器class',
    } as DBWSchema.RdDefineProp;

    const declare = preprocessDeclare(rdProp);

    expect(declare).toMatchSnapshot();
    expect(declare).toMatchObject({
      desc: '容器class',
      id: '641a7052cf44f6204642b6bf',
      key: 'class',
      name: '布局样式',
      types: [
        {
          type: 'text',
        },
      ],
    });
  });

  it('preprocessDeclare text', () => {
    const rdProp = {
      id: '641a7052cf44f6204642b6bf',
      key: 'class',
      types: [
        {
          type: 'text',
        },
      ],
      name: '布局样式',
      desc: '容器class',
    } as DBWSchema.RdDefineProp;

    const declare = preprocessDeclare(rdProp);

    expect(declare).toMatchSnapshot();
    expect(declare).toMatchObject({
      desc: '容器class',
      id: '641a7052cf44f6204642b6bf',
      key: 'class',
      name: '布局样式',
      types: [
        {
          type: 'text',
        },
      ],
    });
  });
});
