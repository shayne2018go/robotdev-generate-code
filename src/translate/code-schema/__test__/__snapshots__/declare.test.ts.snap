// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`declare > preprocessDeclare text 1`] = `
{
  "desc": "容器class",
  "extendPlatform": undefined,
  "id": "641a7052cf44f6204642b6bf",
  "key": "class",
  "name": "布局样式",
  "types": [
    {
      "type": "text",
    },
  ],
}
`;

exports[`declare > preprocessDeclare text 2`] = `
{
  "desc": "容器class",
  "extendPlatform": undefined,
  "id": "641a7052cf44f6204642b6bf",
  "key": "class",
  "name": "布局样式",
  "types": [
    {
      "type": "text",
    },
  ],
}
`;
